routes/auth.js

```diff
@@ -0,0 +1,69 @@
+const router = require("express").Router();
+const bcrypt = require("bcryptjs");
+const jwtService = require("../services/JwtService");
+const Error = require("../models/Error");
+const Joi = require("@hapi/joi");
+
+const userService = require("../services/UserService");
+
+router.post("/login", (req, res) => {
+    const { zprn, password } = req.body;
+    const { error } = validateCredentials(zprn, password);
+    if (error !== undefined) {
+        res.status(400).send(
+            new Error("BAD_REQUEST", error.details[0].message)
+        );
+    } else {
+        userService
+            .getUser(zprn)
+            .then((user) => {
+                bcrypt.compare(password, user.password, (error, match) => {
+                    if (error) {
+                        res.status(500).send(
+                            new Error("INTERNAL_SERVER_ERROR", error)
+                        );
+                    } else {
+                        if (match) {
+                            const token = jwtService.sign({
+                                userID: user.zprn, //? Should userID be added to JWT token?
+                                department: user.department,
+                            });
+
+                            token != null
+                                ? res.status(200).json({ token: token })
```

Comment:
We need to also add a `result` parameter that can have to values only - `failure` and `success`. So that when the api call is returned it is easy to understand whether to consider it as success or failure and read the error. We need to add this param in `Error` as well which will default to `result: "failure"`.

Reply:
Okay